### PR TITLE
Fixes

### LINKED ISSUE #40: Test set contains nodes that are not present in the training graph
The training set doesn't get filtered from the test set after the train-validation set is split up, so there can be nodes in the test set that aren't in the training set, since they are in the validation set.
One fix would be to put all the filtering at the end of the split function.

### PR DESCRIPTION
**Pull request by nomisto**
fixes #40

### CODE REVIEW

**Comment by matthias-samwald** [comment]
Is this ready or are you still working on it? From my side, feel free to merge!
